### PR TITLE
Add python 3.6 support back

### PR DESCRIPTION
**Pull request by fcatalan92**
Turns out that having support for python 3.6 is nice (especially if you are using CentOS7). It looks possible now.

### CODE REVIEW

**Comment by mconcas** [comment]
It is indeed nice and costless to support back py3.6.
I would also really like to avoid something like #104 again. 
Therefore, I suggest to pin the requirements versions, or at least of the annoying packages, e.g. shap and its dependencies (randomly installed thanks to `pip`'s resolution approach).
This, together with the adoption and pinpoitn of `awkward1` and  `uproot 4(?)` would set a stable standard for a while.

What do people think?

File: requirements.txt [hunk 1]
```diff
@@ -4,6 +4,6 @@ matplotlib>=3.3.3
 pandas>=1.1.4
 scikit-learn>=0.23.2
 xgboost>=1.2.1
-shap>=0.37
+shap>=0.38.1
```
Comment (mconcas) [inline requirements.txt:7]: What about
```suggestion
shap=0.38.1
```
?

File: setup.py [hunk 2]
```diff
@@ -84,15 +85,16 @@ def __call__(self):
     # installed. For an analysis of "install_requires" vs pip's requirements files see:
     # https://packaging.python.org/en/latest/requirements.html
     install_requires=["uproot==3.13.1", "matplotlib>=3.3.3", "pandas>=1.1.4", "scikit-learn>=0.23.2",
-                      "xgboost>=1.2.1", "shap>=0.37", "bayesian-optimization>=1.2", "pyarrow>=2.0"],
+                      "xgboost>=1.2.1", "shap>=0.38.1", "bayesian-optimization>=1.2", "pyarrow>=2.0",
```
Comment (mconcas) [inline setup.py:88]: Same here:
```suggestion
                      "xgboost=1.2.1", "shap=0.38.1", "bayesian-optimization>=1.2", "pyarrow=2.0",
```
?

File: setup.py [hunk 3]
```diff
@@ -84,15 +85,16 @@ def __call__(self):
...
+                      "ipython>=6.0.0"],
```
Comment (mconcas) [inline setup.py:89]:
```suggestion
                      "ipython=6.0.0"],
```

File: setup.py [hunk 4]
```diff
@@ -84,15 +85,16 @@ def __call__(self):
...
 
-    python_requires=">=3.7",
+    python_requires=">=3.6",
 
     # List additional groups of dependencies here (e.g. development dependencies). You can install
     # these using the following syntax, for example:
     # $ pip install -e .[dev,test]
     extras_require={
-        "dev": ["pylint>=2.4.4", "flake8>=3.7.9", "pytest>=5.3.4", "twine>=1.11.0", "setuptools>=38.6.0",
+        "dev": ["pylint>=2.4.4", "flake8>=3.7.9", "pytest>=5.3.4", "twine>=1.11.0", "setuptools>=53.0.0",
```
Comment (mconcas) [inline setup.py:97]:
```suggestion
        "dev": ["pylint>=2.4.4", "flake8>=3.7.9", "pytest>=5.3.4", "twine>=1.11.0", "setuptools=53.0.0",
```

**Review by mconcas — CHANGES_REQUESTED** [review]

**Comment by fcatalan92** [comment]
@mconcas I bumped the packages, fixed the versions of the ones you suggested and of those needed to guarantee the python3.6 support. In the end, now almost all of them have the version fixed.
In a separate PR, I can work on migrating to uproot4 and removing some warnings.

**Review by mconcas — APPROVED** [review]

**Review by mconcas — APPROVED** [review]

**Pull request merged**